app/services/distance_calculator_service.rb

```diff
@@ -9,52 +9,38 @@ def initialize(claim, params)
   end
 
   def call
-    return validation_error unless valid_inputs?
-    distance = DistanceCalculatorService::Directions.new(origin, destination).max_distance
-
-    return response(nil) unless distance
-    # NOTE: returned distance is just one way so for the purposes
-    # of the travel expense it needs to be a return distance (doubled up)
-    return response(distance * 2)
+    response.tap do |data|
+      data.value =  DistanceCalculatorService::Directions.new(origin, destination).max_distance unless data.error
+      # NOTE: returned distance is just one way so for the purposes
+      # of the travel expense it needs to be a return distance (doubled up)
+      data.value *= 2 if data.value
+    end
   end
 
   private
 
   attr_reader :claim, :params
-  attr_accessor :validation_error
-
-  def response(value=nil, error: nil)
-    OpenStruct.new(value: value, error: error)
-  end
 
-  def valid_inputs?
-    result = validate_inputs
-    self.validation_error = result if result.error
-    validation_error.nil?
+  def response
+    OpenStruct.new(value: nil, error: validate_inputs)
```

Comment:
openstruct is not very effiecient, prefer struct if attributes are fixed

Reply:
I have updated this to use Struct instead of OpenStruct.